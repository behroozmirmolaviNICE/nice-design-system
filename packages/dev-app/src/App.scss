@import "~@nice-digital/design-system/scss/nice-design-system";

@import "~@nice-digital/nds-forms/scss/forms";

main {
  background: white;
  box-shadow: 0 0 10px rgba(0, 0, 0, .3);
  margin: 3vw;
  padding: 3vw;
  position: relative;
}
