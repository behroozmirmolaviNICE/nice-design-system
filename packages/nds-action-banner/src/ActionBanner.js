// @flow
import React, { Component } from "react";
import "../scss/action-banner.scss";

type ActionBannerProps = {
	title: string,
	modifier: "subtle",
	children: *,
	cta: Element,
	closeable: boolean
};

const ActionBanner = (props: ActionBannerProps) => {

	let classes = ["action-banner"];

	if(props.modifier)
		classes.push(`action-banner--${props.modifier}`);

	if(props.closeable)
		classes.push("action-banner--closeable");

	return <section className={classes.join(" ")}>
		<div className="action-banner__container">
			<div className="action-banner__inner">
				<div className="action-banner__text">
					<h2 className="action-banner__title">
						{props.title}
					</h2>
					<p className="action-banner__intro">
						{props.children}
					</p>
				</div>
				<div className="action-banner__actions">
					{props.cta}
				</div>
				{props.closeable &&
					<button type="button" className="action-banner__close">
						<span className="icon icon--remove" ara-hidden="true"></span>
						<span className="visually-hidden">Close</span>
					</button>
				}
			</div>
		</div>
	</section>;
};

export default ActionBanner;
