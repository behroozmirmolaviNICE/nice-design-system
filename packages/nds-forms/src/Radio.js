// @flow
import React, { Component } from "react";
import classnames from "classnames";
import "../scss/radio.scss";

type RadioProps = {
	group: string,
	label: string,
	value: string,
	inline: boolean,
	error: boolean,
};

export default class Radio extends Component<RadioProps> {

	render() {
		const {error, group, label, value, inline, ...rest} = this.props;
		const classNames = classnames({
			"radio": true,
			"radio--inline": inline,
			"radio--error": error
		});
		const unique = group + "_" + value;
		return (
			<div className={classNames}>
				<input
					className="radio__input"
					name={group}
					type="radio"
					id={unique}
					{...rest}
				/>

				<label
					className="radio__label"
					htmlFor={unique}>
					{label}
				</label>
			</div>
		);
	}
}
