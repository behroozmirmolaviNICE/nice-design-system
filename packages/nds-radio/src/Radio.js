import React, { Component } from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import "../scss/radio.scss";

export class Radio extends Component {
	render() {
		const {
			disabled,
			error,
			group,
			label,
			value,
			inline,
			...rest
		} = this.props;
		if (!value || value === "") return null;
		const classNames = classnames({
			radio: true,
			"radio--inline": inline,
			"radio--error": error
		});
		const unique = `${group}_${value}`;
		return (
			<div className={classNames}>
				<input
					disabled={disabled}
					className="radio__input"
					name={group}
					type="radio"
					id={unique}
					{...rest}
				/>
				<label className="radio__label" htmlFor={unique}>
					{label ? label : value}
				</label>
			</div>
		);
	}
}

Radio.propTypes = {
	disabled: PropTypes.bool,
	error: PropTypes.bool,
	group: PropTypes.string.isRequired,
	inline: PropTypes.bool,
	label: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired
};
