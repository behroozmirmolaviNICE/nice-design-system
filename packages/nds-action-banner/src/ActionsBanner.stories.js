import React from "react";

import { storiesOf } from "@storybook/react";
import { withKnobs } from "@storybook/addon-knobs";

import ActionBanner from "./ActionBanner";

storiesOf("Action banner", module)
	.addDecorator(withKnobs)
	.add("Primary", () => <ActionBanner />);
