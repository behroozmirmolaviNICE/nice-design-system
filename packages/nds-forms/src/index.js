import Radio from "./Radio";
import RadioGroup from "./RadioGroup";
import Textarea from "./Textarea";
import Input from "./Input";
import Fieldset from "./Fieldset";

export { Radio, RadioGroup, Textarea, Input, Fieldset };
