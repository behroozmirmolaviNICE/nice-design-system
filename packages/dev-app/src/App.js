import React, { Component } from "react";
import { Radio, RadioGroup, Textarea, Input, Fieldset, Checkbox } from "@nice-digital/nds-forms";

import "./App.scss";

class App extends Component {
	render() {
		return (
			<div>

				<Fieldset legend="Checkboxes">
					<Checkbox />
				</Fieldset>

				<RadioGroup
					legend="Radio buttons (stacked, default)"
					group="group-1"
					hint="This is some hint text you can read">
					<Radio value="yes" label="Yes please!"/>
					<Radio value="no" label="No thanks!"/>
					<Radio value="maybe" label="Well maybe..? Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ea eaque explicabo, facere harum necessitatibus nulla optio pariatur rem! Aut blanditiis dolores expedita minima mollitia nostrum rem! Accusamus aliquid, distinctio facilis harum illum magni nemo pariatur provident sequi vel, voluptates voluptatum!"/>
				</RadioGroup>

				<RadioGroup legend="Radio buttons inline" group="group-2" hint="This is some hint text you can read" inline>
					<Radio value="yes" label="Yes please!"/>
					<Radio value="no" label="No thanks!"/>
					<Radio value="maybe" label="Well maybe..?"/>
				</RadioGroup>

				<Fieldset legend="Textarea">
					<Textarea
						unique="textarea-1"
						label="Who do you think you are?"
						value="Default value for textarea"/>
					<Textarea
						unique="textarea-1"
						label="Who do you think you are?"
						hint="This is some hint text you can read"
						name="myCommentBox"/>
					<Textarea
						unique="textarea-1"
						label="Who do you think you are?"
						name="mySecondCommentBox"/>
				</Fieldset>

				<Fieldset legend="Inputs">
					<Input
						name="full-name"
						type="text"
						label="Full name"/>
					<Input
						name="email"
						type="email"
						label="Your email address"/>
					<Input
						name="password"
						type="password"
						hint="Your password must contain 4 special characters, 6 numbers, 12 letters and an emoji"
						label="Password"/>
				</Fieldset>

			</div>
		);
	}
}

export default App;
