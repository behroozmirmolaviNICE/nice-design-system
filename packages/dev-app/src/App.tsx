import React, { Component } from "react";
// import Faker from "faker";

import { Checkbox } from "@nice-digital/nds-checkbox";
import "./App.scss";

class App extends Component {
	render() {
		return (
			<main>
				<Checkbox />
			</main>
		);
	}
}

export default App;
