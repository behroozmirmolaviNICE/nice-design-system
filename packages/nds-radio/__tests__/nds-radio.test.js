import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import { Radio } from "../src/Radio";

const props = {
	label: "Yes, please.",
	value: "yes",
	disabled: false,
	error: false,
	inline: false,
	group: "radio-group"
};

describe("Radio", () => {
	it("should render without crashing", () => {
		const wrapper = shallow(<Radio {...props} />);
		expect(wrapper).toHaveLength(1);
	});

	it("should match snapshot with some default attributes", () => {
        const wrapper = shallow(<Radio {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should not render a radio control if it doesn't have a value", () => {
        const localProps = Object.assign({}, props);
        localProps.value = "";
        const wrapper = shallow(<Radio {...localProps} />);
        expect(wrapper.isEmptyRender()).toBe(true);
    });

    it("should prevent actions if the button is disabled", () => {
        const localProps = Object.assign({}, props);
        localProps.disabled = true;
        const wrapper = shallow(<Radio {...localProps} />);
        const input = wrapper.find("input[disabled]");
        expect(input.length).toEqual(1);
    });

    it("should add an error class to the parent if the error prop is true", () => {
        const localProps = Object.assign({}, props);
        localProps.error = true;
        const wrapper = shallow(<Radio {...localProps} />);
        const parent = wrapper.find("div.radio--error");
        expect(parent.length).toEqual(1);
    });

    it("should pass through any additional attributes supplied", () => {
        const localProps = Object.assign({}, props);
        localProps.defaultChecked = true;
        localProps["data-something"] = "test";
        const wrapper = mount(<Radio {...localProps} />);
        const checked = wrapper.find("input[defaultChecked]");
        expect(checked.length).toEqual(1);
        expect(checked.props()["data-something"]).toEqual("test");
    });
});
