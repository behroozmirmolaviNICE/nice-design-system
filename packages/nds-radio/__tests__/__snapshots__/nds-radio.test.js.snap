// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with some default attributes 1`] = `
<div
  className="radio"
>
  <input
    className="radio__input"
    disabled={false}
    id="radio-group_yes"
    name="radio-group"
    type="radio"
  />
  <label
    className="radio__label"
    htmlFor="radio-group_yes"
  >
    Yes, please.
  </label>
</div>
`;
